sprite.cpp

```diff
@@ -99,7 +91,84 @@ void Sprite::draw_trail(Context &ctx) {
 	}
 }
 
-Yonker::Yonker(const Texture *texture, const Point &home) 
+SpriteWiggler::SpriteWiggler()
+	: m_relpos_tendency(0, 0),
+	m_frames_when_tendency_changes(0),
+	m_distance_from_tendency_bias(default_distance_from_tendency_bias.first),
+	m_tendency_distance_from_home_bias(default_tendency_distance_from_home_bias.first),
+	m_wiggle_amount_bias(default_wiggle_amount_bias.first)
+{ }
+
+void SpriteWiggler::wiggle(Context &ctx, const Point &home, Point &relpos, const double &magnitude) {
```

Comment:
Taking small objects like a double by reference may be less efficient than just copying it.  The rule of thumb is to take classes by reference but not primitives.